File: problem_3/problem_3.go
package main

import "fmt"

func addByIndex(listA []int, listB []int) []int {
    sumList := make([]int, len(listA))
    for i := range sumList {
        sumList[i] = listA[i] + listB[i]
    }
    return sumList
}

func main() {
    listA := []int{1, 2, 3}
    listB := []int{4, 5, 6}
    fmt.Println(addByIndex(listA, listB))
}

File: problem_3/problem_3.dart
void main() {
  List<int> addByIndex(List<int> listA, List<int> listB) {
    final sumList = <int>[];
    for (var i = 0; i < listA.length; i++) {
      sumList.add(listA[i] + listB[i]);
    }
    return sumList;
  }

  const listA = [1, 2, 3];
  const listB = [4, 5, 6];
  print(addByIndex(listA, listB));
}
